tests/test_caching.py

```diff
@@ -344,6 +344,41 @@ def test_environment_variable_local_cache_dir(rig: CloudProviderTestRig, tmpdir)
         os.environ["CLOUDPATHLIB_LOCAL_CACHE_DIR"] = original_env_setting
 
 
+def test_environment_variables_force_overwrite(rig: CloudProviderTestRig, tmpdir):
+    # environment instantiation
+    original_env_setting = os.environ.get("CLOUDPATHLIB_FORCE_OVERWRITE_FROM_CLOUD", "")
+
+    try:
```

Comment:
Is there any reason this is a `try`-`finally` instead of pytest's `monkeypatch.setenv`? https://docs.pytest.org/en/latest/how-to/monkeypatch.html

Reply:
Just to follow the pattern of other tests in this part of the suite. I'm going to leave for now, but if we see any issues, we should switch over at some point.